pkg/k8s/deleter/deleter.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package deleter
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/d2iq-labs/capi-runtime-extensions/pkg/constants"
+	"github.com/d2iq-labs/capi-runtime-extensions/pkg/k8s/annotations"
+)
+
+type Deleter struct {
+	cluster *v1beta1.Cluster
+	client  client.Client
+	log     logr.Logger
+}
+
+type objectMetaList []metav1.ObjectMeta
+
+func (ol objectMetaList) asCommaSeparatedString() string {
+	out := ""
+	separator := ""
+	for n := range ol {
+		object := ol[n]
+		out += fmt.Sprintf("%s%s/%s", separator, object.Namespace, object.Name)
+		separator = ", "
+	}
+	return out
+}
```

Comment:
nit: simplify with `strings.Join`

```suggestion
func (ol objectMetaList) asCommaSeparatedString() string {
	names := make([]string, 0, len(ol))
	for _, obj := range ol {
		names = append(names, fmt.Sprintf("%s/%s", obj.Namespace, obj.Name))
	}
	return strings.Join(names, ",")
}
```